src/SWP/Bundle/CoreBundle/Security/Authenticator/ExternalOauthAuthenticator.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SWP\Bundle\CoreBundle\Security\Authenticator;
+
+use Doctrine\ORM\EntityManagerInterface;
+use SWP\Bundle\CoreBundle\Model\UserInterface as CoreUserInterface;
+use SWP\Bundle\CoreBundle\Security\Provider\UserProvider;
+use Symfony\Component\Security\Core\User\UserInterface;
+use KnpU\OAuth2ClientBundle\Security\Authenticator\SocialAuthenticator;
+use KnpU\OAuth2ClientBundle\Client\ClientRegistry;
+use KnpU\OAuth2ClientBundle\Client\OAuth2Client;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+use FOS\UserBundle\Model\UserManagerInterface;
+use Symfony\Component\Security\Core\Security;
+use League\OAuth2\Client\Token\AccessToken;
+
+class ExternalOauthAuthenticator extends SocialAuthenticator
+{
+    /**
+     * @var ClientRegistry
+     */
```

Comment:
docblocks for properties are also not needed if we can see in constructor what type it have.